modules/platforms/cpp/binary/src/impl/binary/binary_object_impl.cpp

```diff
@@ -36,10 +36,23 @@ namespace ignite
                 mem(&mem),
                 start(start),
                 idRslvr(0),
-                metaMgr(metaMgr)
+                metaMgr(metaMgr),
+                binary(false)
             {
                 if (idRslvr)
                     this->idRslvr = idRslvr->Clone();
+
+                int8_t hdr = BinaryUtils::ReadInt8(mem, this->start);
+
+                if (hdr == IGNITE_TYPE_BINARY)
```

Comment:
What if this condition is false? Is it a valid case or should we throw an error?

Reply:
It's OK, the class works on normal objects (hdr = 103) too. Maybe we should throw in other cases though.